pkg/aws/core.go

```diff
@@ -259,18 +259,20 @@ func (d *Driver) InitializeMachine(_ context.Context, request *driver.Initialize
 	if err != nil {
 		return nil, err
 	}
-	instances, err := d.getInstancesFromMachineName(request.Machine.Name, providerSpec, request.Secret)
+	instances, err := d.getMatchingInstancesForMachine(request.Machine, providerSpec, request.Secret)
```

Comment:
can we just pass `providerSpec.Tags` to `getMachineInstancesByTagsAndStatus` function? Inside the function you can extract cluster name and node role.

Reply:
There was actually no reason to pass the entire providerSpec.  Instead only pass the `service`. At present in `InitializeMachine` you are constructing it twice which is not really required.